Utils/utils.ts

```diff
@@ -0,0 +1,89 @@
+import { Browser, BrowserContext, Locator, Page } from "@playwright/test";
+import { operations } from "../enums/operations.enum";
+import { stateEnum } from "../enums/state.enum";
+
+export class utils
+{
+    private readonly page : Page
+
+    constructor(page : Page)
+    {
+        this.page = page;
+    }
+
+    async goto(url : string, nextSelector : string, operation : operations.visible | operations.waitFor = operations.waitFor, timeout? : number) : Promise<boolean>
+    {
+        await this.page.goto(url)
+        if(operation === operations.waitFor)
+        {
+            await this.page.waitForSelector(nextSelector, { state : stateEnum.attached, timeout : timeout})
+        }
+        else
+        {
+            await this.page.goto(url)
+            return await this.isLocatorVisible(nextSelector)
+        }
+
+        return true;
+    }
+
+    async isLocatorVisible(selector : string) : Promise<boolean>
+    {
+        return await (await this.locator(selector)).isVisible();
+    }
+    
+    async locator(selector : string) : Promise<Locator>
+    {
+        return this.page.locator(selector).first();
+    }
+
+    async fill(selector : string, data : string) : Promise<void>
+    {
+        await (await this.locator(selector)).fill(data);
```

Comment:
please wrap all util functions in a try catch, and failure on catch should be descriptive (ex: "failed to fill the selector with error" + err)